File: python/3n-plus-1.py
import json
import os

data = []

for i in range(1, 1000):
    n = i
    list = []

    while (n != 1):
        list.append(n)
        print(n)
        if (n % 2 == 0):
            n = n // 2
        else:
            n = n * 3 + 1
    print("1")
    data.append(list)

with open('data.json', 'w') as f:
    json.dump(data, f, ensure_ascii=False, indent=4)

File: typescript/3n-plus-1.ts
let data:number[] = []
import { writeFile } from 'fs/promises';

for (let i = 1; i < 1000; i++) {
    let n:number = i;
    let list:number[] = [];

    while (n != 1){
        list.push(n);
        console.log(n);
        if (n % 2 == 0){
            n = n / 2;
        }
        else{
            n = n * 3 + 1;
        }
    }
    console.log("1");
    data.push(list);
}

writeFile('data.json', JSON.stringify(data));
